### PR TITLE
feat(FX-3321): add debounce to onchangetext requests in improved search

### PR DESCRIPTION
**Pull request by gkartalis**
The type of this PR is: **Enhancement**

This PR resolves [FX-3321]

### Description

Add debounce to reduce requests

**Before:**
https://user-images.githubusercontent.com/21178754/133793806-109861ca-e173-41a8-bb6f-e5fb2f873e16.mov

**After:**
https://user-images.githubusercontent.com/21178754/133793900-ba882081-2066-464f-b084-68968a8beb43.mov

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- add debounce to reduce requests on improved search - gkartalis

</details>

[FX-3321]: https://artsyproduct.atlassian.net/browse/FX-3321

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (add debounce to reduce requests on improved search - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 76276a020294f2906b8bb8312f2f42a072e4bd86
</p>

File: src/lib/Scenes/Search2/Search2.tsx [hunk 1]
```diff
@@ -41,18 +43,22 @@ const SearchInput: React.FC<SearchInputProps> = ({ currentRefinement, placeholde
...
 
+  const onChangeTextHandler = (queryText: string) => {
+    refine(queryText)
+    trackEvent({
+      action_type: Schema.ActionNames.ARAnalyticsSearchStartedQuery,
+      query: queryText,
+    })
+  }
+
+  const debouncedEventHandler = useMemo(() => debounce(onChangeTextHandler, DEBOUNCE_DELAY), [])
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/Search2.tsx:54]: **thoughts**: I think we can use `useDebounce` [hook](https://github.com/streamich/react-use/blob/master/docs/useDebounce.md) from `react-use` library

Comment (gkartalis) [reply src/lib/Scenes/Search2/Search2.tsx:54]: Will give that a chance! Thanks, didn't know that we had this

Comment (dimatretyak) [reply src/lib/Scenes/Search2/Search2.tsx:54]: Your solution seems easier to use

Comment (gkartalis) [reply src/lib/Scenes/Search2/Search2.tsx:54]: yeah, I spent some time trying to make it work with useDebounce and didn't make it yet, will check it again on Monday 😅

Comment (gkartalis) [reply src/lib/Scenes/Search2/Search2.tsx:56]: no worries! 👍  I learned something from this!

**Review by dimatretyak — APPROVED** [review]

**Comment by MounirDhahri** [comment]
🎉🎉🎉

**Comment by MounirDhahri** [comment]
Hi again, while QAing the app today's morning I noticed that the UX in the search is slightly less convenient than it was although we are only debouncing with 400ms. It got me thinking that debouncing with such a high interval might not be a good idea so is also with a very small interval because that would just never be reached. I would like to reimplement this using throttling to make sure that the user can always tell that there is something happening every short interval. I can create a PR to do it and you can also do it (sorry for the inconvenience)

**Comment by gkartalis** [comment]
Hello @MounirDhahri! Good thought! What about pairing on this sometime today?

**Comment by MounirDhahri** [comment]
Yes, of course, I added some time on your calendar, please feel to reschedule if the time doesn't fit you

**Pull request merged**